geoapps/io/input_file.py

```diff
@@ -401,3 +391,104 @@ def _set_associations(self, d: dict[str, Any]) -> None:
 
             else:
                 continue
+
+
+class UIJson:
+    """Encodes the ui.json format and provides utilities."""
+
+    def __init__(self, ui: dict[str, Any]):
+        self.ui = ui
+
+    @staticmethod
+    def _data(d: dict[str, Any]) -> dict[str, Any]:
+        """Flattens ui.json format to simple key/value pair."""
+        data = {}
+        for k, v in d.items():
+            if isinstance(v, dict):
+                field = "value" if UIJson._truth(d, k, "isValue") else "property"
+                if not UIJson._truth(d, k, "enabled"):
+                    data[k] = None
+                else:
+                    data[k] = v[field]
+            else:
+                data[k] = v
+
+        return data
+
+    def data(self):
+        """Applies _data to self.ui."""
+        return UIJson._data(self.ui)
```

Comment:
Should we not stash the data once we have done it once?

Reply:
I think you're right.. I'll just have to think through what happens to the dictionary outside the scope of UIJson.ui if I edit it inside.  Maybe I should do a deepcopy on init?

Reply:
no wait... that would destroy the uijson format.  This method converts uijson format to simple key/value.

Reply:
maybe you just need to talk me through it

Reply:
You have a  couple of different if...else styles here too that is perhaps making this harder to parse for people?

```python
for k, v in d.items():
+            if isinstance(v, dict):
+                field = "value" if UIJson._truth(d, k, "isValue") else "property"   <----
+                if not UIJson._truth(d, k, "enabled"):
+                    data[k] = None
+                else:
+                    data[k] = v[field]
+            else:
+                data[k] = v
+
```